File: Archiver/app_code/clsATTRIBUTES.vb
Imports Microsoft.VisualBasic
Imports System.Configuration
Imports System
Imports System.Data.SqlClient
Imports System.Collections
Imports System.IO
Imports System.Data.Sql

Public Class clsATTRIBUTES

    '** DIM the selected table columns
    Dim DBARCH As New clsDatabaseARCH

    Dim DMA As New clsDma
    Dim UTIL As New clsUtility
    Dim LOG As New clsLogging

    Dim AttributeName As String = ""
    Dim AttributeDataType As String = ""
    Dim AttributeDesc As String = ""
    Dim AssoApplication As String = ""
    Dim AllowedValues As String = ""

    '** Generate the SET methods
    Public Sub setAttributename(ByRef val As String)
        If Len(val) = 0 Then
            MessageBox.Show("SET: Field 'Attributename' cannot be NULL.")
            Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        AttributeName = val
    End Sub

    Public Sub setAttributedatatype(ByRef val As String)
        If Len(val) = 0 Then
            MessageBox.Show("SET: Field 'Attributedatatype' cannot be NULL.")
            Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        AttributeDataType = val
    End Sub

    Public Sub setAttributedesc(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        AttributeDesc = val
    End Sub

    Public Sub setAssoapplication(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        AssoApplication = val
    End Sub

    Public Sub setAllowedvalues(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        AllowedValues = val
    End Sub

    '** Generate the GET methods
    Public Function getAttributename() As String
        If Len(AttributeName) = 0 Then
            MessageBox.Show("GET: Field 'Attributename' cannot be NULL.")
            Return ""
        End If
        Return UTIL.RemoveSingleQuotes(AttributeName)
    End Function

    Public Function getAttributedatatype() As String
        If Len(AttributeDataType) = 0 Then
            MessageBox.Show("GET: Field 'Attributedatatype' cannot be NULL.")
            Return ""
        End If
        Return UTIL.RemoveSingleQuotes(AttributeDataType)
    End Function

    Public Function getAttributedesc() As String
        Return UTIL.RemoveSingleQuotes(AttributeDesc)
    End Function

    Public Function getAssoapplication() As String
        Return UTIL.RemoveSingleQuotes(AssoApplication)
    End Function

    Public Function getAllowedvalues() As String
        Return UTIL.RemoveSingleQuotes(AllowedValues)
    End Function

    '** Generate the Required Fields Validation method
    Public Function ValidateReqData() As Boolean
        If AttributeName.Length = 0 Then Return False
        If AttributeDataType.Length = 0 Then Return False
        Return True
    End Function

    '** Generate the Validation method
    Public Function ValidateData() As Boolean
        If AttributeName.Length = 0 Then Return False
        If AttributeDataType.Length = 0 Then Return False
        Return True
    End Function

    '** Generate the INSERT method
    Public Function Insert() As Boolean
        Dim b As Boolean = False
        Dim s As String = ""
        s = s + " INSERT INTO Attributes("
        s = s + "AttributeName,"
        s = s + "AttributeDataType,"
        s = s + "AttributeDesc,"
        s = s + "AssoApplication,"
        s = s + "AllowedValues) values ("
        s = s + "'" + AttributeName + "'" + ","
        s = s + "'" + AttributeDataType + "'" + ","
        s = s + "'" + AttributeDesc + "'" + ","
        s = s + "'" + AssoApplication + "'" + ","
        s = s + "'" + AllowedValues + "'" + ")"
        Return DBARCH.ExecuteSqlNewConn(s, False)

    End Function

    '** Generate the UPDATE method
    Public Function Update(ByVal WhereClause As String) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        If Len(WhereClause) = 0 Then Return False

        s = s + " update Attributes set "
        's = s + "AttributeName = '" + getAttributename() + "'" + ", "
        s = s + "AttributeDataType = '" + getAttributedatatype() + "'" + ", "
        s = s + "AttributeDesc = '" + getAttributedesc() + "'" + ", "
        s = s + "AssoApplication = '" + getAssoapplication() + "'" + ", "
        s = s + "AllowedValues = '" + getAllowedvalues() + "'"
        WhereClause = " " + WhereClause
        s = s + WhereClause
        Return DBARCH.ExecuteSqlNewConn(s, False)
    End Function

    '** Generate the SELECT method
    Public Function SelectRecs() As SqlDataReader
        Dim b As Boolean = False
        Dim s As String = ""
        Dim rsData As SqlDataReader
        s = s + " SELECT "
        s = s + "AttributeName,"
        s = s + "AttributeDataType,"
        s = s + "AttributeDesc,"
        s = s + "AssoApplication,"
        s = s + "AllowedValues "
        s = s + " FROM Attributes"
        '** s=s+ "ORDERBY xxxx"
        Dim CS As String = DBARCH.setConnStr()     'DBARCH.getGateWayConnStr(gGateWayID) : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
        Return rsData
    End Function

    '** Generate the Select One Row method
    Public Function SelectOne(ByVal WhereClause As String) As SqlDataReader
        Dim b As Boolean = False
        Dim s As String = ""
        Dim rsData As SqlDataReader
        s = s + " SELECT "
        s = s + "AttributeName,"
        s = s + "AttributeDataType,"
        s = s + "AttributeDesc,"
        s = s + "AssoApplication,"
        s = s + "AllowedValues "
        s = s + " FROM Attributes"
        s = s + WhereClause
        '** s=s+ "ORDERBY xxxx"
        Dim CS As String = DBARCH.setConnStr()     'DBARCH.getGateWayConnStr(gGateWayID) : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
        Return rsData
    End Function

    '** Generate the DELETE method
    Public Function Delete(ByVal WhereClause As String) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        If Len(WhereClause) = 0 Then Return False

        WhereClause = " " + WhereClause

        s = " Delete from Attributes"
        s = s + WhereClause

        b = DBARCH.ExecuteSqlNewConn(s, False)
        Return b

    End Function

    '** Generate the Zeroize Table method
    Public Function Zeroize() As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        s = s + " Delete from Attributes"

        b = DBARCH.ExecuteSqlNewConn(s, False)
        Return b

    End Function

    '** Generate Index Queries
    Public Function cnt_PK36(ByVal AttributeName As String) As Integer

        Dim B As Integer = 0
        Dim TBL As String = "Attributes"
        Dim WC As String = "Where AttributeName = '" + AttributeName + "'"

        B = DBARCH.iGetRowCount(TBL, WC)

        Return B
    End Function     '** cnt_PK36

    '** Generate Index ROW Queries
    Public Function getRow_PK36(ByVal AttributeName As String) As SqlDataReader

        Dim rsData As SqlDataReader = Nothing
        Dim TBL As String = "Attributes"
        Dim WC As String = "Where AttributeName = '" + AttributeName + "'"

        rsData = DBARCH.GetRowByKey(TBL, WC)

        If rsData.HasRows Then
            Return rsData
        Else
            Return Nothing
        End If
    End Function     '** getRow_PK36

    ''' Build Index Where Caluses
    Public Function wc_PK36(ByVal AttributeName As String) As String

        Dim WC As String = "Where AttributeName = '" + AttributeName + "'"

        Return WC
    End Function     '** wc_PK36

    '** Generate the SET methods

End Class

File: Archiver/app_code/clsATTRIBUTES.cs
using global::System.Data.SqlClient;
using System.Windows.Forms;
using global::Microsoft.VisualBasic;
using MODI;

namespace EcmArchiver
{
    public class clsATTRIBUTES
    {

        // ** DIM the selected table columns
        private clsDatabaseARCH DBARCH = new clsDatabaseARCH();
        private clsDma DMA = new clsDma();
        private clsUtility UTIL = new clsUtility();
        private clsLogging LOG = new clsLogging();
        private string AttributeName = "";
        private string AttributeDataType = "";
        private string AttributeDesc = "";
        private string AssoApplication = "";
        private string AllowedValues = "";

        // ** Generate the SET methods
        public void setAttributename(ref string val)
        {
            if (Strings.Len(val) == 0)
            {
                MessageBox.Show("SET: Field 'Attributename' cannot be NULL.");
                return;
            }

            val = UTIL.RemoveSingleQuotes(val);
            AttributeName = val;
        }

        public void setAttributedatatype(ref string val)
        {
            if (Strings.Len(val) == 0)
            {
                MessageBox.Show("SET: Field 'Attributedatatype' cannot be NULL.");
                return;
            }

            val = UTIL.RemoveSingleQuotes(val);
            AttributeDataType = val;
        }

        public void setAttributedesc(ref string val)
        {
            val = UTIL.RemoveSingleQuotes(val);
            AttributeDesc = val;
        }

        public void setAssoapplication(ref string val)
        {
            val = UTIL.RemoveSingleQuotes(val);
            AssoApplication = val;
        }

        public void setAllowedvalues(ref string val)
        {
            val = UTIL.RemoveSingleQuotes(val);
            AllowedValues = val;
        }

        // ** Generate the GET methods
        public string getAttributename()
        {
            if (Strings.Len(AttributeName) == 0)
            {
                MessageBox.Show("GET: Field 'Attributename' cannot be NULL.");
                return "";
            }

            return UTIL.RemoveSingleQuotes(AttributeName);
        }

        public string getAttributedatatype()
        {
            if (Strings.Len(AttributeDataType) == 0)
            {
                MessageBox.Show("GET: Field 'Attributedatatype' cannot be NULL.");
                return "";
            }

            return UTIL.RemoveSingleQuotes(AttributeDataType);
        }

        public string getAttributedesc()
        {
            return UTIL.RemoveSingleQuotes(AttributeDesc);
        }

        public string getAssoapplication()
        {
            return UTIL.RemoveSingleQuotes(AssoApplication);
        }

        public string getAllowedvalues()
        {
            return UTIL.RemoveSingleQuotes(AllowedValues);
        }

        // ** Generate the Required Fields Validation method
        public bool ValidateReqData()
        {
            if (AttributeName.Length == 0)
                return false;
            if (AttributeDataType.Length == 0)
                return false;
            return true;
        }

        // ** Generate the Validation method
        public bool ValidateData()
        {
            if (AttributeName.Length == 0)
                return false;
            if (AttributeDataType.Length == 0)
                return false;
            return true;
        }

        // ** Generate the INSERT method
        public bool Insert()
        {
            bool b = false;
            string s = "";
            s = s + " INSERT INTO Attributes(";
            s = s + "AttributeName,";
            s = s + "AttributeDataType,";
            s = s + "AttributeDesc,";
            s = s + "AssoApplication,";
            s = s + "AllowedValues) values (";
            s = s + "'" + AttributeName + "'" + ",";
            s = s + "'" + AttributeDataType + "'" + ",";
            s = s + "'" + AttributeDesc + "'" + ",";
            s = s + "'" + AssoApplication + "'" + ",";
            s = s + "'" + AllowedValues + "'" + ")";
            return DBARCH.ExecuteSqlNewConn(s, false);
        }

        // ** Generate the UPDATE method
        public bool Update(string WhereClause)
        {
            bool b = false;
            string s = "";
            if (Strings.Len(WhereClause) == 0)
                return false;
            s = s + " update Attributes set ";
            // s = s + "AttributeName = '" + getAttributename() + "'" + ", "
            s = s + "AttributeDataType = '" + getAttributedatatype() + "'" + ", ";
            s = s + "AttributeDesc = '" + getAttributedesc() + "'" + ", ";
            s = s + "AssoApplication = '" + getAssoapplication() + "'" + ", ";
            s = s + "AllowedValues = '" + getAllowedvalues() + "'";
            WhereClause = " " + WhereClause;
            s = s + WhereClause;
            return DBARCH.ExecuteSqlNewConn(s, false);
        }

        // ** Generate the SELECT method
        public SqlDataReader SelectRecs()
        {
            bool b = false;
            string s = "";
            var rsData = default(SqlDataReader);
            s = s + " SELECT ";
            s = s + "AttributeName,";
            s = s + "AttributeDataType,";
            s = s + "AttributeDesc,";
            s = s + "AssoApplication,";
            s = s + "AllowedValues ";
            s = s + " FROM Attributes";
            // ** s=s+ "ORDERBY xxxx"
            string CS = DBARCH.setConnStr();     // DBARCH.getGateWayConnStr(gGateWayID) : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
            return rsData;
        }

        // ** Generate the Select One Row method
        public SqlDataReader SelectOne(string WhereClause)
        {
            bool b = false;
            string s = "";
            var rsData = default(SqlDataReader);
            s = s + " SELECT ";
            s = s + "AttributeName,";
            s = s + "AttributeDataType,";
            s = s + "AttributeDesc,";
            s = s + "AssoApplication,";
            s = s + "AllowedValues ";
            s = s + " FROM Attributes";
            s = s + WhereClause;
            // ** s=s+ "ORDERBY xxxx"
            string CS = DBARCH.setConnStr();     // DBARCH.getGateWayConnStr(gGateWayID) : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
            return rsData;
        }

        // ** Generate the DELETE method
        public bool Delete(string WhereClause)
        {
            bool b = false;
            string s = "";
            if (Strings.Len(WhereClause) == 0)
                return false;
            WhereClause = " " + WhereClause;
            s = " Delete from Attributes";
            s = s + WhereClause;
            b = DBARCH.ExecuteSqlNewConn(s, false);
            return b;
        }

        // ** Generate the Zeroize Table method
        public bool Zeroize()
        {
            bool b = false;
            string s = "";
            s = s + " Delete from Attributes";
            b = DBARCH.ExecuteSqlNewConn(s, false);
            return b;
        }

        // ** Generate Index Queries
        public int cnt_PK36(string AttributeName)
        {
            int B = 0;
            string TBL = "Attributes";
            string WC = "Where AttributeName = '" + AttributeName + "'";
            B = DBARCH.iGetRowCount(TBL, WC);
            return B;
        }     // ** cnt_PK36

        // ** Generate Index ROW Queries
        public SqlDataReader getRow_PK36(string AttributeName)
        {
            SqlDataReader rsData = null;
            string TBL = "Attributes";
            string WC = "Where AttributeName = '" + AttributeName + "'";
            rsData = DBARCH.GetRowByKey(TBL, WC);
            if (rsData.HasRows)
            {
                return rsData;
            }
            else
            {
                return null;
            }
        }     // ** getRow_PK36

        /// Build Index Where Caluses
        public string wc_PK36(string AttributeName)
        {
            string WC = "Where AttributeName = '" + AttributeName + "'";
            return WC;
        }     // ** wc_PK36

        // ** Generate the SET methods

    }
}